src/TestSuite/Fixture/TestFixture.php

```diff
@@ -159,7 +169,11 @@ protected function _schemaFromFields()
         }
         if (!empty($this->fields['_constraints'])) {
             foreach ($this->fields['_constraints'] as $name => $data) {
-                $this->_schema->addConstraint($name, $data);
+                if ($isSqlite || $data['type'] !== Table::CONSTRAINT_FOREIGN) {
```

Comment:
Instead of a class check, could this be a method like `$connection->supportsDynamicConstraints()`? This would let other datasources/drivers implement it as necessary.